sdk/keyvault/Azure.Security.KeyVault.Keys/src/RecoverDeletedKeyOperation.cs

```diff
@@ -2,30 +2,33 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
 using Azure.Core;
-using Azure.Core.Pipeline;
 
 namespace Azure.Security.KeyVault.Keys
 {
     /// <summary>
     /// A long-running operation for <see cref="KeyClient.StartRecoverDeletedKey(string, CancellationToken)"/> or <see cref="KeyClient.StartRecoverDeletedKeyAsync(string, CancellationToken)"/>.
     /// </summary>
-    public class RecoverDeletedKeyOperation : Operation<KeyVaultKey>
+    public class RecoverDeletedKeyOperation : Operation<KeyVaultKey>, IOperation
     {
         private static readonly TimeSpan s_defaultPollingInterval = TimeSpan.FromSeconds(2);
```

Comment:
still used?

Reply:
Yeah, in the default `WaitForCompletionAsync`. At first I removed it, but we had chosen 2s by default because 1s was too often and hammering the service unnecessarily. I didn't want to change behavior here.

Reply:
Is there a reason you didn't switch to OperationInternal.WaitForCompletion?

Reply:
Because I didn't notice it existed. 🙂 I'll do so.

Reply:
Ah, actually I can't if I don't switch to using `OperationInternal<T>`, which I [mentioned](https://github.com/Azure/azure-sdk-for-net/pull/25593#discussion_r760672502) isn't necessarily practical or necessary.